### PR TITLE
task for hathi exclude list

### PR DESCRIPTION
**Pull request by banukutlu**
#252

### CODE REVIEW

File: lib/tasks/hathi_trust.rake [hunk 1]
```diff
@@ -71,9 +72,25 @@ namespace :hathitrust do
     print `cat hathi_single_oclc.csv hathi_multi_oclc_split.csv > hathi_full_dedupe_with_headers.csv`
   end
 
+  desc 'Process exclude list'
+  task :process_excludes, [:overlap_file] do |_task, args|
+    excludes = ['2168941']
```
Comment (cdmo) [inline lib/tasks/hathi_trust.rake:77]:
```suggestion
    # This is a list of catkeys that we've been asked for one reason or another to not show a HathiTrust link for at all   
    excludes = ['2168941']
```

File: lib/tasks/hathi_trust.rake [hunk 2]
```diff
@@ -71,9 +72,26 @@ namespace :hathitrust do
...
+    CSV.read(args[:overlap_file], { col_sep: "\t", headers: true, header_converters: :symbol }).each do |row|
+      data << row unless excludes.include? row[:local_id]
+    end
+
+    CSV.open('overlap_filtered.csv', 'wb') do |csv|
+      data.each do |row|
+        csv << row
+      end
+    end
+  end
```
Comment (cdmo) [inline lib/tasks/hathi_trust.rake:90]: what do you think about using `delete_if` to condense and outdent code a bit?

```suggestion
  desc 'Process exclude list'
  task :process_excludes, [:overlap_file] do |_task, args|
    # This is a list of catkeys that we've been asked for one reason or another to not show a HathiTrust link for at all
    excludes = ['2168941']
    
    overlaps_csv = CSV.read(args[:overlap_file], { col_sep: "\t", headers: true, header_converters: :symbol })
    filtered_overlaps_csv_file = CSV.open('overlap_filtered.csv', 'wb')
    filtered_overlaps_csv_file << overlaps_csv.delete_if {|row| excludes.include? row[:local_id] }
  end
```

Comment (banukutlu) [reply lib/tasks/hathi_trust.rake:90]: I just tried this but I got errors, the overlap_filtered.csv looks like this ```","449171,1213268,mono,deny,ic
","873986340,12132680,mono,deny,ic
","789661399,12132681,mono,,
","853313642,12132682,mono,,
","869764394,12132683,mono,deny,ic```

did it thru for you? 🤔

Comment (cdmo) [reply lib/tasks/hathi_trust.rake:90]: I see yeah, I made a mistake there with the writing. Sorry. I looked at the output of using `delete_if` which was the main thing I was providing feedback about and that matched what I expected, but I didn't test my final modifications.

Ok, so this is right I think and what do you think?:

```rb
overlaps_csv = CSV.read(args[:overlap_file], col_sep: "\t", headers: true, header_converters: :symbol)
                  .delete_if {|row| excludes.include? row[:local_id] }
CSV.open 'overlap_filtered.cs, 'wb' do |csv|
  overlaps_csv.each { |row| csv << row }
end
```

So `CSV::Table`'s shovel takes a row and appends. So I assumed that `CSV`'s shovel took a `CSV::Table`. Not so fast!

Comment (banukutlu) [reply lib/tasks/hathi_trust.rake:90]: I tried using delete_if myself as well but did not work properly and also even tried CSV::Table as well. Got it to work with CSV::Table but it did not well so I did not use it.

Did it work with delete_if here?

Comment (cdmo) [reply lib/tasks/hathi_trust.rake:90]: Hmm, yeah it worked. I did test it before suggesting. Here's the pry lines:

```
[32] pry(main)> zzz = CSV.read('ignorethis_hathi/overlap_20200914_psu.tsv', col_sep: "\t", headers: true, header_converters: :symbol).delete_if {|row| excludes.include? row[:local_id] }
=> #<CSV::Table mode:col_or_row row_count:2521844>
```

**Review by cdmo — APPROVED** [review]
👍

**Pull request merged**